clustpy/deep/_abstract_deep_clustering_algo.py

```diff
@@ -0,0 +1,47 @@
+from clustpy.deep._utils import set_torch_seed
+from sklearn.base import BaseEstimator, ClusterMixin
+from sklearn.utils import check_random_state
+import numpy as np
+import torch
+
+
+class _AbstractDeepClusteringAlgo(BaseEstimator, ClusterMixin):
```

Comment:
Added this class which can be extended to included commonly used functionalities